src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -134,8 +139,9 @@ private static ParquetProperties createParquetProperties() {
         // pages is the same.
         // The quick fix is to effectively disable the page size/row limit
         // to always have one page per chunk until server side is generalised.
-        .withPageSize((int) Constants.MAX_CHUNK_SIZE_IN_BYTES * 2)
+        .withPageSize(pageSizeLimit)
```

Comment:
do you know the page size limit is before or after compression?

Reply:
Page size limit is for compressed data, i.e. after compression.

Reply:
thanks, what if they see the size is bigger than the size limit after compression? Will they uncompressed it and split?

Reply:
> thanks, what if they see the size is bigger than the size limit after compression?

IIUC there is no size check after compression, only before to decide when to start a page.